graphql/src/execution/execution.rs

```diff
@@ -156,15 +197,22 @@ pub fn execute_root_selection_set(
     ctx: &ExecutionContext<impl Resolver>,
     selection_set: &q::SelectionSet,
 ) -> Result<BTreeMap<String, q::Value>, Vec<QueryExecutionError>> {
+    // Cache the cache key to not have to calculate it twice - once for lookup
+    // and once for insert.
+    let mut key: Option<QueryHash> = None;
+
     if CACHED_SUBGRAPH_IDS.contains(&ctx.query.schema.id) {
+        // Calculate the hash outside of the lock
+        key = Some(cache_key(ctx, selection_set));
+
         // Check if the response is cached.
         let mut cache = QUERY_CACHE.lock().unwrap();
 
         // Remove expired entries.
         cache.notify_iter();
 
         // Peek because we want even hot entries to invalidate after the expiry period.
-        if let Some(response) = cache.peek(&cache_key(ctx, selection_set)) {
+        if let Some(response) = cache.peek(key.as_ref().unwrap()) {
             debug!(ctx.logger, "Query cache hit");
```

Comment:
I second the comment from the previous PR: we should remove this log statement, since it can cause a lot of logging activity under load

Reply:
Done. Logs: the (not so?) silent killer.